src/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -490,6 +478,13 @@ public IThreadPoolWorkItem Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool
                     }
                     c--;
                 }
+
+                if (callback != null)
+                {
+                    // Set the search start to the next queue, if it is currently set to no preference
+                    i = (i < maxIndex) ? i + 1 : 0;
+                    Interlocked.CompareExchange(ref WorkStealingQueueList.QueueSerachStart, i, -1);
```

Comment:
Idea was to skip the search of all the threads before it as they have been determined empty. However; thinking about it, if only one thread is producing work then the move after this thread would cause a full scan of every thread again - maybe random is best...

Reply:
I see, makes sense